pkg/scheduler/plugins/elasticquota/quota_overuse_revoke.go

```diff
@@ -0,0 +1,272 @@
+		klog.Infof("Quota used continue large than runtime, prepare trigger evict, quotaName:%v,"+
+			"overUseContinueCount:%v, config:%v", monitor.quotaName, monitor.overUsedContinueCount,
+			monitor.continueOverUsedCountTriggerEvict)
+		monitor.overUsedContinueCount = 0
+		return true
+	}
+	return false
+}
+
+func (monitor *QuotaOverUsedGroupMonitor) GetToRevokePodList(quotaName string) []*v1.Pod {
+	quotaInfo := monitor.groupQuotaManger.GetQuotaInfoByName(quotaName)
+	if quotaInfo == nil {
+		return nil
+	}
+
+	runtime := quotaInfo.GetRuntime()
+	used := quotaInfo.GetUsed()
+	oriUsed := used.DeepCopy()
+
+	// order pod from low priority -> high priority
+	priPodCache := treeset.NewWith(PodPriorityComparator)
+	for _, pod := range quotaInfo.GetPodRunningCache() {
+		priPodCache.Add(pod)
+	}
+
+	// first try revoke all until used <= runtime
+	tryAssignBackPodCache := make([]*v1.Pod, 0)
+	iter := priPodCache.Iterator()
+	for iter.Next() {
+		if shouldBreak, _ := quotav1.LessThanOrEqual(used, runtime); shouldBreak {
+			break
+		}
+
+		pod := iter.Value().(*v1.Pod)
+		podReq, _ := resource.PodRequestsAndLimits(pod)
+		used = quotav1.Subtract(used, podReq)
+		tryAssignBackPodCache = append(tryAssignBackPodCache, pod)
+	}
+
+	// means should evict all
+	if lessThanOrEqual, _ := quotav1.LessThanOrEqual(used, runtime); !lessThanOrEqual {
+		for _, pod := range tryAssignBackPodCache {
+			klog.Infof("pod should be revoked by QuotaOverUsedMonitor, pod:%v, quotaName:%v"+
+				"used:%v, runtime:%v", pod.Name, quotaName, oriUsed, runtime)
+		}
+		return tryAssignBackPodCache
+	}
+
+	//try assign back from high->low
+	realRevokePodCache := make([]*v1.Pod, 0)
+	for index := len(tryAssignBackPodCache) - 1; index >= 0; index-- {
+		func() {
+			pod := tryAssignBackPodCache[index]
+			podRequest, _ := resource.PodRequestsAndLimits(pod)
+
+			used = quotav1.Add(used, podRequest)
+			if canAssignBack, _ := quotav1.LessThanOrEqual(used, runtime); !canAssignBack {
+				used = quotav1.Subtract(used, podRequest)
+				realRevokePodCache = append(realRevokePodCache, pod)
+			}
+		}()
+	}
+	for _, pod := range realRevokePodCache {
+		klog.Infof("pod should be evict by QuotaOverUseGroupMonitor, pod:%v, quotaName:%v,"+
+			"used:%v, runtime:%v", pod.Name, quotaName, oriUsed, runtime)
+	}
+	return realRevokePodCache
+}
+
+func PodPriorityComparator(a, b interface{}) int {
+	lhs := a.(*v1.Pod)
+	rhs := b.(*v1.Pod)
+
+	if corev1.PodPriority(lhs) > corev1.PodPriority(rhs) {
+		return 1
+	} else if corev1.PodPriority(lhs) < corev1.PodPriority(rhs) {
+		return -1
+	}
+
+	return 0
+}
+
+type QuotaOverUsedRevokeController struct {
+	clientSet                         clientset.Interface
+	groupQuotaManger                  *core.GroupQuotaManager
+	group2MonitorMapLock              sync.RWMutex
+	group2MonitorMap                  map[string]*QuotaOverUsedGroupMonitor
+	continueOverUsedCountTriggerEvict int64
```

Comment:
`continueOverUsedCountTriggerEvict` may not be easy for users to understand. It might be simpler to use the concept of time, e.g. delayEvictTime?

Reply:
fixed